server/src/test/java/io/crate/auth/JwtAuthenticationIntegrationTest.java

```diff
@@ -83,9 +84,15 @@ public class JwtAuthenticationIntegrationTest extends IntegTestCase {
      *    ]
      * }
      */
-    private static final BiConsumer<io.netty.handler.codec.http.HttpRequest, JsonGenerator> jwkRequestHandler =
-        (io.netty.handler.codec.http.HttpRequest msg, JsonGenerator generator) -> {
+
+    private final class RequestHandler implements BiConsumer<io.netty.handler.codec.http.HttpRequest, JsonGenerator> {
+
+        AtomicInteger numberOfInvocation = new AtomicInteger(0);
+
+        @Override
+        public void accept(io.netty.handler.codec.http.HttpRequest httpRequest, com.fasterxml.jackson.core.JsonGenerator generator) {
             try {
```

Comment:
We had a version with unit test as far as I remember but I'm fine with integration test - I guess idea is to test all components together? 

Especially given that `HostBasedAuthentication` must hold a single CHM instance for all `JwtAuthenticationMethod` instances created per -connection?

Reply:
The unit test version only tests the concurrent hashmap caching for the `JWTAuthenticationMethod`. This integration test is needed anyway, and is testing both variants.